builder/dimension/lookup.go

```diff
@@ -37,6 +37,11 @@ func (l *Lookup) SetOutputName(outputName string) *Lookup {
 	return l
 }
 
+func (l *Lookup) SetDimension(dimensionName string) *Lookup {
```

Comment:
better name it dimension instead of dimensionName since it's what we've done everywhere (just like druid own codebase `@JsonProperty("dimension") String dimension,` https://github.com/apache/druid/blob/master/processing/src/main/java/org/apache/druid/query/dimension/LookupDimensionSpec.java)

Reply:
Thanks for the detailed review, @jbguerraz 

I completely agree with your thoughts. I hadn't thought of specifying it until I looked at the Java source code. I created a DimensionSpec interface and modified the Fields variable of cadinality. Please check if it matches your idea. 

```go
type DimensionSpec interface {
	Dimension
	GetDimension() string
	GetOutputName() string
	GetOutputType() types.OutputType
	GetExtractionFn() ExtractionFn // Deprecated
}
```